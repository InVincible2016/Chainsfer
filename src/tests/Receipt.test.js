import moment from 'moment'
import Button from '@material-ui/core/Button'
import update from 'immutability-helper'
import { CopyToClipboard } from 'react-copy-to-clipboard'
import Tooltip from '@material-ui/core/Tooltip'

import Receipt from '../components/ReceiptComponent'

const initialProps = {
  backToHome: () => {},
  cryptoSelection: 'ethereum',
  txFee: {
    costInBasicUnit: '21000000000000',
    costInStandardUnit: '0.000021',
    gas: '21000',
    price: '1000000000'
  },
  receipt: {
    cryptoType: 'ethereum',
    destination: 'abc@gmail.com',
    sendTimestamp: '1558022548',
    sendTxHash: '0xe1d591e4d59057550d4f41abd066b211c547c7ae643b9da1de2cc7fbb2659ce9',
    sender: 'abc@gmail.com',
    sendingId: 'e1981151-b951-467f-81e4-e9f87dfa44a8',
    transferAmount: '0.001'
  },
  password: '123456',
  sendTime: moment.unix('1558022548').utc().format('MMM Do YYYY, HH:mm:ss')
}

jest.mock('react-router-dom', () => () => ({
  Link: 'Link'
}))

let wrapper
describe('ReceiptComponent render:', () => {
  beforeEach(() => {
    wrapper = mount(
      <Receipt {...initialProps} />
    )
  })

  it('Receipt render:', () => {
    expect(toJson(wrapper.render())).toMatchSnapshot()
  })
})

describe('ReceiptComponent interactions:', () => {
  beforeEach(() => {
    wrapper = mount(
      <Receipt {...initialProps} />
    )
  })

  it('Back to home btn', () => {
    let mockBackToHome = jest.fn()
    wrapper.setProps(update(initialProps, {
      backToHome: mockBackToHome
    })
    )
    wrapper.find(Button).filter('#back').simulate('click')
    expect(mockBackToHome.mock.calls.length).toEqual(1)
  })

  it('copy password', () => {
    // Next line is necessary to access state
    // https://github.com/airbnb/enzyme/issues/431#issuecomment-362318989
    wrapper = mount(shallow(<Receipt {...initialProps} />).get(0))

    wrapper.find(CopyToClipboard).props().onCopy()
    // force update
    wrapper.update()
    expect(wrapper.state('copied')).toEqual(true)
    expect(wrapper.find(Tooltip).prop('open')).toEqual(true)
  })
})
