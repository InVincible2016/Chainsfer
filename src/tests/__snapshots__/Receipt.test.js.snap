// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiptComponent render: Receipt render: 1`] = `
<div
  class="MuiGrid-container-11 MuiGrid-direction-xs-column-14 MuiGrid-align-items-xs-center-19 MuiGrid-justify-xs-center-28"
>
  <div
    class="MuiGrid-item-12"
  >
    <div
      class="MuiPaper-root-108 MuiPaper-elevation1-111 MuiPaper-rounded-109 ReceiptComponent-receiptPaper-6"
    >
      <div
        class="MuiGrid-container-11 MuiGrid-direction-xs-column-14 MuiGrid-justify-xs-center-28"
      >
        <div
          class="MuiGrid-item-12"
        >
          <div
            class="MuiGrid-item-12"
          >
            <div
              class="MuiGrid-container-11 MuiGrid-direction-xs-column-14 MuiGrid-align-items-xs-center-19 MuiGrid-justify-xs-center-28"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-135 ReceiptComponent-checkCircleIcon-7"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <h6
                class="MuiTypography-root-144 MuiTypography-h6-161 MuiTypography-alignCenter-167 ReceiptComponent-title-1"
              >
                Transfer Sent
              </h6>
              <p
                class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignCenter-167 ReceiptComponent-transferId-2"
              >
                Transfer ID: e1981151-b951-467f-81e4-e9f87dfa44a8
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewSubtitle-3"
            >
              From
            </p>
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewContent-4"
            >
              abc@gmail.com
            </p>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewSubtitle-3"
            >
              To
            </p>
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewContent-4"
            >
              abc@gmail.com
            </p>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewSubtitle-3"
            >
              Amount
            </p>
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewContent-4"
            >
              0.001 ETH
            </p>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewSubtitle-3"
            >
              Transaction Fee
            </p>
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewContent-4"
            >
              0.000021 ETH
            </p>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewSubtitle-3"
            >
              Sent on
            </p>
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewContent-4"
            >
              May 16th 2019, 16:02:28
            </p>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewSubtitle-3"
            >
              Security Answer
            </p>
            <p
              class="MuiTypography-root-144 MuiTypography-body1-153 MuiTypography-alignLeft-166 ReceiptComponent-reviewContent-4"
            >
              123456
              <button
                class="MuiButtonBase-root-194 MuiIconButton-root-188 ReceiptComponent-iconBtn-10"
                tabindex="0"
                title="Copied"
                type="button"
              >
                <span
                  class="MuiIconButton-label-193"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-135 MuiSvgIcon-fontSizeSmall-142"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
                    />
                  </svg>
                </span>
              </button>
            </p>
          </div>
          <div
            class="MuiGrid-item-12 ReceiptComponent-reviewItem-5"
          >
            <p
              class="MuiTypography-root-144 MuiTypography-body2-152 MuiTypography-alignLeft-166 ReceiptComponent-informReceiverText-8"
            >
              Please inform receiver the security answer to complete the transaction.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-12 ReceiptComponent-btnSection-9"
  >
    <div
      class="MuiGrid-container-11 MuiGrid-spacing-xs-24-35 MuiGrid-justify-xs-center-28"
    >
      <div
        class="MuiGrid-item-12"
      >
        <button
          class="MuiButtonBase-root-194 MuiButton-root-197 MuiButton-contained-208 MuiButton-containedPrimary-209 MuiButton-raised-211 MuiButton-raisedPrimary-212 MuiButton-sizeLarge-221 MuiButton-fullWidth-222"
          id="back"
          tabindex="0"
          to="/"
          type="button"
        >
          <span
            class="MuiButton-label-198"
          >
            Back to Home
          </span>
          <span
            class="MuiTouchRipple-root-223"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
